### DISCUSSION TITLE
HA restart triggers action

### DISCUSSION THREAD

**Discussion by tjorim** [ITSpecialist111/ai_automation_suggester#26] [Ideas]
When restarting HA (or reloading the integration) it gets triggered and creates a new persistent notification, even when the set scan frequency hasn't passed.

**Comment by ITSpecialist111** [comment]
Hi @tjorim,

Thank you for bringing this up!

I’m pleased to let you know that in the latest release (v1.04), we’ve addressed this issue. The integration now checks whether the configured scan_frequency has passed before triggering a new persistent notification when restarting Home Assistant or reloading the integration. This should prevent unnecessary notifications from being generated when no new scan is required.

**Comment by ITSpecialist111** [comment]
I can see this has come back after the code review. Will look to resolve this with better frequency settings as they may chew through GPT tokens.

**Comment by ITSpecialist111** [comment]
Testing....

V1.07 Now Provides these asks:

Removal of Automatic Scheduling:

Removed automatic scanning and suggestion generation due to complexities with triggering and user preferences.
Enhanced Stability: This change improves code stability and allows users to implement custom automation triggers that suit their individual needs.

Expanded AI Provider Support:

Added Support for Multiple LLM Models:
Google LLM Models: Now supports Google's AI models for generating automation suggestions.
Local AI Models: Introduced support for local models like LocalAI and Ollama for users who prefer privacy and do not rely on internet connectivity.

Custom LLM Variants:
Users can now choose custom AI model variants from higher models (e.g., OpenAI's o1-preview), providing more flexibility and control over the AI processing.
Enhanced Manual Trigger Service:

Integration Selection: Users can now select which AI provider integration to trigger when manually initiating the AI analysis.
Custom System Prompt: Allows users to override the built-in system prompt for more granular and personalized suggestions.
German Translation Added:

Global Reach: Introduced German language support to make the integration accessible to a wider audience.

**Comment by ITSpecialist111** [comment]
This is still happening during rebooting in V1.08. It's because Home Assistant is reloading the integration. I'll continue to workout how this might be prevented in future releases. For now I will move this issue to upgrade requests.

**Reply by tjorim** [reply]
HA restart is my main concern. E.g. when updating to HA beta versions or to apply updates from custom integrations. But thanks for the effort already put into this!

**Comment by Zixim** [comment]
I'm running into this , before I go troubleshooting - can someone please confirm if this was ever resolved ?

**Reply by ITSpecialist111** [reply]
It still triggers on reboot yes. It's the way the integration is monitoring the APIs

**Comment by Zixim** [comment]
That isn't really workable, because it's spending tokens to check the API.
For instance, I was doing some re-configuring in HAOS (unrelated to this integration)... did a couple of restarts & eventually restored from backup.
After all of this, the free plan on Mistral AI was exhausted.
